### PR TITLE
Use golangci-lint in CI

### PR DESCRIPTION
**Pull request by cfergeau**
This patch series fixes the various issues reported by golangci-lint, and then enable it in the CI.
Some of these issues were actual bugs.

### CODE REVIEW

**Comment by praveenkumar** [comment]
@cfergeau All the commits are related to the `golangci-lint` can we squash those since these are anyway required and it's not like we are changing the function signature (iirc) ?

**Comment by cfergeau** [comment]
> @cfergeau All the commits are related to the `golangci-lint` can we squash those since these are anyway required and it's not like we are changing the function signature (iirc) ?

I don't think they should be squashed, review is much easier when you can tell why a given change is being made (which would be harder with a single commit), and if one of these commit is wrong, bisection/reverting will be easier too.

File: .circleci/config.yml [hunk 1]
```diff
@@ -71,7 +74,9 @@ jobs:
         command: podman info
     - run:
         name: Make build_docs
-        command: make build_docs
+        # force GOPATH to a sane value as the image we are using has an older go version
+        # with a 'go env GOPATH' output which causes make failures
+        command: make GOPATH="/home/circleci/go" build_docs
```
Comment (praveenkumar) [inline .circleci/config.yml]: 👍

File: .travis.yml [hunk 2]
```diff
@@ -6,4 +6,5 @@ go:
 script:
 - make
 - make fmtcheck
+- make lint
```
Comment (praveenkumar) [inline .travis.yml]: 👍

File: Makefile [hunk 3]
```diff
@@ -69,7 +70,7 @@ vendor:
 	GO111MODULE=on go mod vendor
 
 # Get binappend
-binappend:
+$(GOPATH)/bin/binappend-cli:
```
Comment (praveenkumar) [inline Makefile]: 👍

**Pull request merged**